### DISCUSSION TITLE
Checking if the Planetscale database is in sync with the schema

### DISCUSSION THREAD

**Discussion by ErkoKnoll** [prisma/prisma#15154] [Q&A] [answered]
Is there an official way of checking if the database is in sync with the schema? We're working with Planetscale and have to perform separate Deploy Requests in Planetscale to perform migrations, which currently relies on devs remembering to merge DR requests before merging PRs for the actual Prisma code. And if they forget to do so, things will blow up horribly, until the schema is synced with the database. I'd like to perform this check in deploy time to prevent any PRs from being deployed if the target database has not been brought to sync with the prisma schema that is about to be deployed.

I hoped that I could do it with `npx prisma db push` command, hoping it to blow up complaining about not being able to execute DDL statements in Planetscale prod branches, which it does. However, it also blows up and complains about DDL statements even if there is nothing to push. This would have been most straight forward approach to halt any deployments if the schema and the database were out of sync, but unfortunately it also blows up even if they are.

**Comment by janpio** [comment, marked as answer]
Did you have a look at https://www.prisma.io/docs/reference/api-reference/command-reference#migrate-diff? You should probably be able to use `--exit-code` to achieve what you want.

**Reply by ErkoKnoll** [reply]
Thanks @janpio, this is exactly what I was missing.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
